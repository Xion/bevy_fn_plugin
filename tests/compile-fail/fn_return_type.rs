#[macro_use] extern crate bevy_fn_plugin;

use bevy::prelude::*;

#[bevy_plugin]
fn has_return_type(_: &mut App) -> usize { 0 }

fn main() {}
