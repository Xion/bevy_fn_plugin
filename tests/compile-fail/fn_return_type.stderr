error: Invalid plugin function signature

         = help: Function should take a single argument of the type `&mut bevy::app::App`

 --> tests/compile-fail/fn_return_type.rs:6:1
  |
6 | fn has_return_type(_: &mut App) -> usize { 0 }
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `bevy::prelude::*`
 --> tests/compile-fail/fn_return_type.rs:3:5
  |
3 | use bevy::prelude::*;
  |     ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
